indico/modules/events/registration/models/items.py

```diff
@@ -437,6 +437,12 @@ def view_data(self):
     def _get_default_log_data(self):
         return {'Section ID': self.id}
 
+    @make_interceptable
+    def can_see(self, user, is_management):
+        if not user and self.registration_form.require_login:
```

Comment:
I don't think this new check is necessary, is it?

Reply:
That is because "user" is added in arguments,
otherwise the code won't be understandable, WHY adding "user" if no reference ?

Reply:
```py
"""Check whether the user can see the section.

By default it only matters whether the section is viewed while in
management mode or not, but plugins may intercept this to perform
additional checks based on the user who's viewing the section.
"""
```

Problem solved.

---

However, I think there's a much easier option which I did not realize in your initial PR version since you already used `session.user` in there: Since none of the indico code makes use of the user, you can completely ignore it. And in your plugin you are of course free to use `session.user`.

I would just rename it from `can_see` (since those methods typically do take a user) to `is_visible` (and then not having the user arg there at all).

Reply:
Updated as suggested.